### PR TITLE
replay: add support for dcecompressing ZST log files

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

**Comment by sshane** [comment]
@deanlee we're getting this merged soon, is this good to go?

**Comment by deanlee** [comment]
Yes, I've tested it and it's ready to go. However, there is still an issue with the selfdrive/PR comments. It can't find #include <zstd.h>. It seems we need to update the Docker image to include the zstd library. Do you have any ideas on how to resolve this?

**Comment by sshane** [comment]
Looks like it's building the docker image from the base branch, which is expected to fail here. Hopefully we can clean up this job very soon, but I wouldn't make it a blocker for this PR. It should pass once we merge.

File: tools/replay/util.cc [hunk 1]
```diff
@@ -318,6 +319,36 @@ std::string decompressBZ2(const std::byte *in, size_t in_size, std::atomic<bool>
...
+  while (input.pos < input.size && !(abort && *abort)) {
+      ZSTD_outBuffer output = { outputBuffer.data(), bufferSize, 0 };
+
+      size_t result = ZSTD_decompressStream(dctx, &output, &input);
+      if (ZSTD_isError(result)) {
+        rWarning("decompressZST error: content is corrupt");
+        break;
+      }
+
+      decompressedData.append(outputBuffer.data(), output.pos);
```
Comment (sshane) [inline tools/replay/util.cc:341]: 2 spaces?

File: tools/replay/logreader.cc [hunk 2]
```diff
@@ -7,8 +7,13 @@
 
 bool LogReader::load(const std::string &url, std::atomic<bool> *abort, bool local_cache, int chunk_size, int retries) {
   std::string data = FileReader(local_cache, chunk_size, retries).read(url, abort);
-  if (!data.empty() && url.find(".bz2") != std::string::npos)
-    data = decompressBZ2(data, abort);
+  if (!data.empty()) {
+    if (url.find(".bz2") != std::string::npos) {
+      data = decompressBZ2(data, abort);
+    } else if (url.find(".zst") != std::string::npos) {
+      data = decompressZST((std::byte *)data.data(), data.size(), abort);
```
Comment (sshane) [inline tools/replay/logreader.cc:14]: Heads up we may remove the file extensions soon, so it should detect from the file header like the python LogReader does.

**Comment by deanlee** [comment]
The PR is ready, suggested changes have been applied.

**Comment by sshane** [comment]
ty!

**Pull request merged**